evm/logic/system.py

```diff
@@ -147,10 +159,17 @@ def __call__(self, computation):
         if child_computation.error:
             computation.stack.push(0)
         else:
-            computation.gas_meter.return_gas(child_computation.gas_meter.gas_remaining)
             computation.stack.push(contract_address)
+        computation.gas_meter.return_gas(child_computation.gas_meter.gas_remaining)
```

Comment:
Why was this change necessary?

Reply:
The previous code with this nested into the `else` block was just an optimization since the amount of gas to return in the case of an error was always zero.  Now with the advent of revert, we should *always* return whatever gas is leftover.